### PR TITLE
Add qcost function and test

### PR DESCRIPTION
**Pull request by aidanheerdegen**
Add a `qcost` function to calculate how many SUs a job will consume.

Also added some basic tests.

### CODE REVIEW

**Comment by aidanheerdegen** [comment]
@ScottWales Can you take a look? I had already put the queue info stuff in before you pointed me to `uqstat`. We can chuck it in `qtools.py` I guess? Feels like this is getting awfully close to becoming a `conda` package, but not sure if it is worth the effort or not.

File: qcost [hunk 1]
```diff
@@ -0,0 +1,139 @@
+#!/usr/bin/env python3
```
Comment (ScottWales) [inline qcost:1]: check the shbang of uqstat, that uses conda's python always

File: qcost [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+def extract_num_unit(s):
+    # Match a number (possibly floating point 100.00 style) and a unit
+    try:
+        size, unit = re.findall(r'(\d+.\d+|\d+)\s*(\D*)$',s)[0]
+    except:
+        print('Failed to match size string: ',s)
+        sys.exit()
+    return float(size), unit
+
+def pretty_size(n,pow=0,b=1024,u='B',pre=['']+[p for p in'KMGTPEZY']):
```
Comment (ScottWales) [inline qcost:25]: for info no need to change - dask has dask.utils.format_bytes and dask.utils.parse_bytes which can be handy for this.

File: qcost [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+    """
+
+    global queue_info
+
+    seconds_in_an_hour = 3600.
+
+    # Normalise time multiplier to 1 hour
+    timefactor = to_seconds(walltime) / seconds_in_an_hour
+
+    if queue_info is None:
```
Comment (ScottWales) [inline qcost:94]: I don't think the singleton is that necessary here - queue_info is small. Another way to do this is to decorate load_data with `@functools.lru_cache`

**Review by ScottWales — COMMENTED** [review]
Happy with this, only thing that needs to be changed is the shbang line for the script

File: qcost (hunk 1 above)
Comment (aidanheerdegen) [reply qcost:1]: Tah

File: qcost (hunk 2 above)
Comment (aidanheerdegen) [reply qcost:25]: Good to know. Tah.

File: qcost (hunk 3 above)
Comment (aidanheerdegen) [reply qcost]: Agreed. I just removed it for simplicity.

**Pull request merged**